rust/platform/triple_mappings.bzl

```diff
@@ -188,11 +194,32 @@ _SYSTEM_TO_STDLIB_LINKFLAGS = {
 }
 
 def cpu_arch_to_constraints(cpu_arch):
+    if cpu_arch not in _CPU_ARCH_TO_BUILTIN_PLAT_SUFFIX:
+        fail("CPU architecture \"{}\" is not supported by rules_rust".format(cpu_arch))
+
     plat_suffix = _CPU_ARCH_TO_BUILTIN_PLAT_SUFFIX[cpu_arch]
 
-    if not plat_suffix:
+    return ["@platforms//cpu:{}".format(plat_suffix)]
+
+def cpu_arch_constraints_from_triple(triple):
```

Comment:
Why not update `cpu_arch_to_constraints` to match the pattern of `abi_to_constraints`? I don't particularly care which pattern is used but I would like there to be a consistent API.

Reply:
`cpu_arch_to_constraints` is also used in 
https://github.com/bazelbuild/rules_rust/blob/7f2dd433ac0f0fee6ee1ef90544a648640a66526/rust/platform/platform.bzl#L36
which I don't think it has access to the full abi / triple.

I can update `cpu_arch_constraints_from_triple` to match `abi_to_constraints` if that is what you are asking for.

Reply:
Yeah, I would like that :)

Reply:
Updated with the new interface.